docarray/array/doc_dict/doc_dict.py

```diff
@@ -0,0 +1,131 @@
+            super().update(other)
+        if isinstance(other, DocList):
+            super().update(DocDict.from_doc_list(other))
+        elif isinstance(other, Mapping):
+            super().update(other)
+        elif isinstance(other, Iterable):
+            super().update(
+                DocDict.from_doc_list(DocList.__class_getitem__(self.doc_type)(other))
+            )
+        else:
+            raise ValueError(f'Cannot update DocDict with {other}')
+
+    def _get_data_column(
+        self: T,
+        field: str,
+    ) -> Union[T, Dict[Any, Union[BaseDoc, 'TorchTensor', 'NdArray']]]:
+
+        field_type = self.__class__.doc_type._get_field_type(field)
+
+        if (
+            not is_union_type(field_type)
+            and self.__class__.doc_type.__fields__[field].required
+            and isinstance(field_type, type)
+            and issubclass(field_type, BaseDoc)
+        ):
+            return DocDict.__class_getitem__(field_type)(  # todo skip validation
+                **{key: getattr(doc, field) for key, doc in self.items()},
+            )
+        else:
+            return {key: getattr(doc, field) for key, doc in self.items()}
+
+    def _set_data_column(
+        self: T,
+        field: str,
+        values: Mapping[str, Any],
+    ):
+        """Set all Documents in this `DocDict` using the passed values
+
+        :param field: name of the fields to set
+        :values: the values to set at the `DocList` level
+        """
+        if values.keys() != self.keys():
+            raise ValueError(
+                f'Keys of the values {values.keys()} do not match the keys of the DocDict {self.keys()}'
+            )
+
+        for key, value in values.items():
+            setattr(self[key], field, value)
+
+    @classmethod
+    def from_protobuf(cls: Type[T], pb_msg: 'DocDictProto') -> T:
+        """create a DocDict from a protobuf message"""
+        return cls(
+            **{
+                key: cls.doc_type.from_protobuf(doc_proto)
+                for key, doc_proto in pb_msg.docs.items()
+            }
+        )
+
+    def to_protobuf(self) -> 'DocDictProto':
+        """Convert `DocDict` into a Protobuf message"""
+        from docarray.proto import DocDictProto
+
+        return DocDictProto(docs={key: doc.to_protobuf() for key, doc in self.items()})
+
+    @classmethod
+    def from_json(
+        cls: Type[T],
+        file: Union[str, bytes, bytearray],
+    ) -> T:
+        """Deserialize JSON strings or bytes into a `DocList`.
+
+        :param file: JSON object from where to deserialize a `DocList`
+        :return: the deserialized `DocList`
+        """
+        json_docs = orjson.loads(file)
+        return cls(**{key: cls.doc_type(**doc) for key, doc in json_docs.items()})
+
+    def to_json(self) -> bytes:
+        """Convert the object into JSON bytes. Can be loaded via `.from_json`.
+        :return: JSON serialization of `DocList`
+        """
+        return orjson_dumps(self)
```

Comment:
Is the goal to:

1- abstract away some part of the code to avoid to repeat our self.
or 
2- provide a base abstract class that will work as an Interface to show to the user what is available.


I thought about this while implementing and basically:

I think 1 is not needed bc most of the io code is cannot be factorized between different classes (or very few part of it and it will just be a pain to maintain)

for 2 I agree that we need it, But since we don't support json stuff on DocVec for now I was thinking that it is not really needed for now but will def'ly be when we harmonized all of the export

Reply:
I was thinking of 2, but makese sense, let's do that after IO for DocVec is merged